### PR TITLE
fix: Show correct 'no results' message on workspace filters

### LINKED ISSUE #2042: Empty workspace search results should clarify that there are no results
image
Create Workspace is the wrong next step if the user has workspaces.

Related #2041

### PR DESCRIPTION
**Pull request by f0ssel**
Fixes https://github.com/coder/coder/issues/2042.

Open to suggestions on what to display here, right now there's currently no description text. 

chrome_99vpCmpq2c

### CODE REVIEW

File: site/src/pages/WorkspacesPage/WorkspacesPageView.tsx [hunk 1]
```diff
@@ -51,19 +53,29 @@ export const WorkspacesPageView: FC<WorkspacesPageViewProps> = ({ loading, works
...
-                  cta={
-                    <Link underline="none" component={RouterLink} to="/workspaces/new">
-                      <Button startIcon={<AddCircleOutline />}>{Language.createButton}</Button>
-                    </Link>
-                  }
-                />
-              </TableCell>
-            </TableRow>
+            <>
+              {query === "owner:me" || query === "" ? (
```
Comment (vapurrmaid) [inline site/src/pages/WorkspacesPage/WorkspacesPageView.tsx:57]: I'd make these constants and re-use those constants here and in the form menu. Maybe in `utils/workspaces.ts`

**Review by greyscaled — APPROVED** [review]

**Comment by vapurrmaid** [comment]
Can we adjust the storybook so that there's a case for each the paths?

See what I mean here: https://www.chromatic.com/test?appId=624de63c6aacee003aa84340&id=629e690bcbc710004ad776c9

**Comment by f0ssel** [comment]
Will need some guidance on the story stuff, I'm looking at https://github.com/coder/coder/blob/main/site/src/pages/WorkspacesPage/WorkspacesPageView.stories.tsx and I'm pretty lost on how to add my query stuff.

**Comment by presleyp** [comment]
@f0ssel happy to get on discord sometime today and figure out the storybook stuff

**Comment by BrunoQuaresma** [comment]
Probably some of these conflicts are related to this PR https://github.com/coder/coder/pull/2097 basically what I made to cause this was to move all the logic from WorkspacePage to the WorkspacePageView so we could easily test that using Storybook. Please, let me know if you need help to solve the conflict.

**Pull request merged**